daemon/images/image_delete.go

```diff
@@ -66,7 +67,16 @@ func (i *ImageService) ImageDelete(ctx context.Context, imageRef string, options
 	start := time.Now()
 	records := []imagetypes.DeleteResponse{}
 
-	img, err := i.GetImage(ctx, imageRef, backend.GetImageOpts{})
+	var platform *ocispec.Platform
+	switch len(options.Platforms) {
+	case 0:
+	case 1:
+		platform = &options.Platforms[0]
+	default:
+		return nil, errdefs.InvalidParameter(errors.New("multiple platforms are not supported"))
```

Comment:
This is probably something we want to implement in future? (if so, perhaps we should create a tracking-ticket for that work)